Mathlib/GroupTheory/PresentedGroup.lean

```diff
@@ -53,6 +61,23 @@ theorem closure_range_of (rels : Set (FreeGroup α)) :
     FreeGroup.closure_range_of, ← MonoidHom.range_eq_map]
   exact MonoidHom.range_top_of_surjective _ (QuotientGroup.mk'_surjective _)
 
+@[induction_eliminator]
+theorem induction_on {rels : Set (FreeGroup α)}  {C : PresentedGroup rels → Prop}
+    (x : PresentedGroup rels) (H : ∀ z, C (mk rels z)) : C x :=
+  Quotient.inductionOn' x H
+
+theorem generated_by (rels : Set (FreeGroup α)) (H : Subgroup (PresentedGroup rels))
+    (h : ∀ j : α, PresentedGroup.of j ∈ H) : ∀ x : (PresentedGroup rels), x ∈ H := by
+  intro x
+  apply induction_on x
+  intro z
+  apply FreeGroup.induction_on (C := fun z => ⟦z⟧ ∈ H) _ (one_mem H) (fun _ => h _)
+    fun _ => (Subgroup.inv_mem_iff H).mpr
+  intro _ _ h1 h2
```

Comment:
Can these use the `induction` tactic? If not, why not?

Reply:
both can! thanks, I was being silly